src/game/Objects/Unit.cpp

```diff
@@ -473,9 +470,9 @@ void Unit::SendMovementPacket(uint16 opcode, bool includingSelf)
     SendMovementMessageToSet(std::move(data), includingSelf);
 }
 
-void Unit::ResetAttackTimer(WeaponAttackType type)
+void Unit::ResetAttackTimer(WeaponAttackType type, bool compensateDiff/*= false*/)
 {
-    m_attackTimer[type] = uint32(GetAttackTime(type) * m_modAttackSpeedPct[type]);
+    m_attackTimer[type] = uint32(GetAttackTime(type) * m_modAttackSpeedPct[type]) + ((compensateDiff) ? std::min(m_attackTimer[type], 0) : 0);

```

Comment:
I suggest to add a separate if statement below instead of doing it like that.

Reply:
If I try to do it like this,. m_attackTimer was already set to the attackspeed in the line above it, so this will always add 0
```
m_attackTimer[type] = uint32(GetAttackTime(type) * m_modAttackSpeedPct[type])
if  (compensateDiff) 
    m_attackTimer[type] += std::min(m_attackTimer[type], 0)
// m_attackTimer was already set to the attack speed above, so this will always add min(m_attackTimer, 0) = 0
```

So I can first save it then add the saved value after setting to 0 like this:

```
int32 savedAttackTimer = std::min(m_attackTimer[type], 0)
m_attackTimer[type] = uint32(GetAttackTime(type) * m_modAttackSpeedPct[type])
if  (compensateDiff) 
    m_attackTimer[type] += savedAttackTimer 
```
Is this method better?

Reply:
What about
```
if (compensateDiff && m_attackTimer[type] < 0)
    m_attackTimer[type] += int32(GetAttackTime(type) * m_modAttackSpeedPct[type])
else
    m_attackTimer[type] = int32(GetAttackTime(type) * m_modAttackSpeedPct[type]);
```